app/assets/javascripts/initializers/initializeBodyData.js

```diff
@@ -49,7 +49,7 @@ function fetchBaseData() {
         }, 400);
       } else {
         // Ensure user data is not exposed if no one is logged in
-        document.body.dataset.user = undefined;
+        delete document.body.dataset.user;
```

Comment:
Good stuff. I was going to suggest this or `document.body.removeAttribute('data-user');` since it being undefined doesn't really make sense as a value. 👏🏻